### PR TITLE
Setup prow 

### PR DESCRIPTION
**Pull request by roshanirathi**

### CODE REVIEW

**Comment by spectro-prow** [comment]
@roshanirathi: Cannot trigger testing until a trusted user reviews the PR and leaves an `/ok-to-test` message.

<details>

In response to [this](https://github.com/spectrocloud/cluster-api-provider-maas/pull/3#issuecomment-889699503):

>/test cluster-api-provider-maas_build


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by spectro-prow** [comment]
@roshanirathi: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
cluster-api-provider-maas_build | 305c07064570d022070433f2ed6959d337ed0f45 | [link](http://34.120.121.77/view/gcs/capi-prow-artifacts/pr-logs/pull/spectrocloud_cluster-api-provider-maas/3/cluster-api-provider-maas_build/1421013989055270912) | `/test cluster-api-provider-maas_build`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by spectro-prow** [comment]
@roshanirathi: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
cluster-api-provider-maas_build | 305c07064570d022070433f2ed6959d337ed0f45 | [link](http://34.120.121.77/view/gcs/capi-prow-artifacts/pr-logs/pull/spectrocloud_cluster-api-provider-maas/3/cluster-api-provider-maas_build/1421036512757157888) | `/test cluster-api-provider-maas_build`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/cluster-api-provider-maas/pull/3#" title="Author self-approved">roshanirathi</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Fcluster-api-provider-maas).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/cluster-api-provider-maas/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by spectro-prow** [comment]
@roshanirathi: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
cluster-api-provider-maas_build | a49bccfd266735d2abdd3fb60bc830eb33f3ab4f | [link](http://34.120.121.77/view/gcs/capi-prow-artifacts/pr-logs/pull/spectrocloud_cluster-api-provider-maas/3/cluster-api-provider-maas_build/1421040413489238016) | `/test cluster-api-provider-maas_build`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by spectro-prow** [comment]
@roshanirathi: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
cluster-api-provider-maas_build | a9ddfc1dd46a9d051f75d0d36dcb975f906ea65e | [link](http://34.120.121.77/view/gcs/capi-prow-artifacts/pr-logs/pull/spectrocloud_cluster-api-provider-maas/3/cluster-api-provider-maas_build/1421043181826347008) | `/test cluster-api-provider-maas_build`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by spectro-prow** [comment]
@roshanirathi: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
cluster-api-provider-maas_image_build | 0888506ab853165da8e1be7d054e8beae52577a3 | [link](http://34.120.121.77/view/gcs/capi-prow-artifacts/pr-logs/pull/spectrocloud_cluster-api-provider-maas/3/cluster-api-provider-maas_image_build/1421051738395250688) | `/test cluster-api-provider-maas_image_build`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by spectro-prow** [comment]
@roshanirathi: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
cluster-api-provider-maas_image_build | 0888506ab853165da8e1be7d054e8beae52577a3 | [link](http://34.120.121.77/view/gcs/capi-prow-artifacts/pr-logs/pull/spectrocloud_cluster-api-provider-maas/3/cluster-api-provider-maas_image_build/1421051738395250688) | `/test cluster-api-provider-maas_image_build`
cluster-api-provider-maas_build | 8ff120ff82a772c30c29bc4f173f061a631fee24 | [link](http://34.120.121.77/view/gcs/capi-prow-artifacts/pr-logs/pull/spectrocloud_cluster-api-provider-maas/3/cluster-api-provider-maas_build/1421054254902153216) | `/test cluster-api-provider-maas_build`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by spectro-prow** [comment]
@roshanirathi: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
cluster-api-provider-maas_image_build | a6827de64d04e9e46936eeceeebb76a305bc20b2 | [link](http://34.120.121.77/view/gcs/capi-prow-artifacts/pr-logs/pull/spectrocloud_cluster-api-provider-maas/3/cluster-api-provider-maas_image_build/1421054758260576256) | `/test cluster-api-provider-maas_image_build`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: prow/release_build.sh
Comment (zulfilee) [inline prow/release_build.sh]: Not required as we dont have sonarqube

File: prow/functions.sh
Comment (zulfilee) [inline prow/functions.sh]: This should be spectro-images-public

**Review by zulfilee — CHANGES_REQUESTED** [review]

File: prow/release_build.sh
Comment (roshanirathi) [reply prow/release_build.sh]: updated code

File: prow/functions.sh
Comment (roshanirathi) [reply prow/functions.sh]: updated

**Pull request merged**
